src/main/java/org/sagebionetworks/bridge/services/OAuthProviderService.java

```diff
@@ -242,7 +281,31 @@ protected OAuthAccessGrant jsonToGrant(JsonNode node) {
         grant.setProviderUserId(providerUserId);
         return grant;
     }
-    
+
+    // Helper method which takes the JSON body of an Introspect request and parses out the set of scopes.
+    protected List<String> jsonToScopeList(JsonNode node) {
+        // RFC7662 defines scope as a space-delimited string. However, FitBit uses a non-standard format that looks
+        // like "{SLEEP=READ, HEARTRATE=READ, ACTIVITY=READ}".
+        String scopeString = node.get(SCOPE_PROP_NAME).textValue();
+        String[] scopePairArray = SCOPE_PAIR_SPLIT_PATTERN.split(scopeString);
+
+        // scopePairArray now looks like "SLEEP=READ", "HEARTRATE=READ", "ACTIVITY=READ". We only care about the scope
+        // name, ie everything before the =. After the equal is always READ or READWRITE (usually READ), and we only
+        // care about reading, so we can ignore it.
+        List<String> scopeList = new ArrayList<>();
+        for (String scopePair : scopePairArray) {
+            // Note that because of the way we split strings, we may have some empty strings. Skip those.
+            if (scopePair.isEmpty()) {
+                continue;
+            }
+
+            int equalIdx = scopePair.indexOf('=');
+            scopeList.add(scopePair.substring(0, equalIdx));
+        }
+
+        return scopeList;
+    }
+
```

Comment:
Are we asking for all the scopes that Fitbit tells us exist? If a partner asks us to get scopes A, B, and C, and Fitbit has A, B, C, and D, do we ask for A-C or A-D? Just asking for clarification...

Reply:
RFC7662 defines this as scopes associated with the token, not all possible scopes. I'll add to the comment to clarify and add an example of the full response in the comments.

Reply:
Further clarification: Bridge doesn't ask OAuth for specific scopes. Scopes are defined by the call to OAuth authorization, which happens in the app. Scopes are locked in at this time, and Bridge Server cannot change them. What Bridge Server does is query the token for what scopes are associated with it.

Reply:
Okay great, thanks.